src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -393,17 +393,17 @@ protected override void Dispose(bool disposing)
                 return activitySamplingResult;
             }
 
-            return PropagateOrIgnoreData(options.Parent.TraceId);
+            return PropagateOrIgnoreData(options.Parent.TraceId, options.Parent.IsRemote);
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static ActivitySamplingResult PropagateOrIgnoreData(ActivityTraceId traceId)
+        private static ActivitySamplingResult PropagateOrIgnoreData(ActivityTraceId traceId, bool isParentRemote)
```

Comment:
We could probably explore updating the method signature here to something like this:
`private static ActivitySamplingResult PropagateOrIgnoreData(in ActivityContext parentContext)` and pass in `options.Parent` to this method. `ActivityContext` is a `readonly struct` so using the `in` modifier might give some minor perf gains.